source/Playnite/HdrUtilities.cs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        /// <summary>
+        /// Determines if HDR is enabled on the primary display
+        /// </summary>
+        /// <returns>True if HDR is enabled on the primary display, false if unknown or not supported</returns>
+        public static bool? IsHdrEnabled()
+        {
+            DISPLAYCONFIG_PATH_TARGET_INFO? targetInfo = GetPrimaryDisplayTargetInfo();
+            if (!targetInfo.HasValue)
+            {
+                return false;
+            }
+
+            (_, bool? isHdrEnabled) = GetHdrStatus(targetInfo.Value);
+            return isHdrEnabled;
+        }
+
+        /// <summary>
+        /// Enables/disables HDR on the primary display
+        /// </summary>
+        /// <param name="enable">True if enabling HDR, false if disabling HDR</param>
+        public static void EnableHdr(bool enable)
+        {
+            DISPLAYCONFIG_PATH_TARGET_INFO? targetInfo = GetPrimaryDisplayTargetInfo();
+            if (!targetInfo.HasValue)
+            {
+                return;
+            }
+
+            if (IsHdrSupported())
+            {
+                //Enable or Disable HDR
+                var newColorInfo = new DISPLAYCONFIG_SET_ADVANCED_COLOR_STATE()
+                {
+                    header =
+                    {
+                        type = DISPLAYCONFIG_DEVICE_INFO_TYPE.DISPLAYCONFIG_DEVICE_INFO_SET_ADVANCED_COLOR_STATE,
+                        adapterId = targetInfo.Value.adapterId,
+                        id = targetInfo.Value.id,
+                        size = (uint)Marshal.SizeOf<DISPLAYCONFIG_SET_ADVANCED_COLOR_STATE>()
+                    }
+                };
+
+                newColorInfo.enableAdvancedColor = enable;
+
+                User32.DisplayConfigSetDeviceInfo(ref newColorInfo.header);
+            }
+        }
+
+        private static bool IsPrimaryDisplayMode(DISPLAYCONFIG_MODE_INFO mode)
+        {
+            return
+                mode.infoType == DISPLAYCONFIG_MODE_INFO_TYPE.DISPLAYCONFIG_MODE_INFO_TYPE_SOURCE &&
+                mode.modeInfo.sourceMode.position.x == 0 &&
+                mode.modeInfo.sourceMode.position.y == 0;
+        }
+
+        private static DISPLAYCONFIG_PATH_TARGET_INFO? GetPrimaryDisplayTargetInfo()
+        {
+            long result = User32.GetDisplayConfigBufferSizes(QUERY_DEVICE_CONFIG_FLAGS.QDC_ONLY_ACTIVE_PATHS, out uint pathCount, out uint modeCount);
+            if (result != WinError.ERROR_SUCCESS)
+            {
+                return null;
+            }
+
+            var paths = new DISPLAYCONFIG_PATH_INFO[pathCount];
+            var modes = new DISPLAYCONFIG_MODE_INFO[modeCount];
+
+            result = User32.QueryDisplayConfig(QUERY_DEVICE_CONFIG_FLAGS.QDC_ONLY_ACTIVE_PATHS, ref pathCount, paths, ref modeCount, modes, IntPtr.Zero);
+            if (result != WinError.ERROR_SUCCESS)
+            {
+                return null;
+            }
+
+            DISPLAYCONFIG_MODE_INFO primaryDisplayMode = modes
```

Comment:
Please use `var` instead where possible.

Reply:
Will do. Although this might be something you wish to clarify in coding conventions for this repo. There are counterpoints to this, specifically that explicit usage of the type may be better if the right-hand side of the `=` doesn't provide a clear indication on the type.

Reply:
done